appcues/src/main/java/com/appcues/data/mapper/experience/ExperienceMapper.kt

```diff
@@ -87,10 +85,12 @@ internal class ExperienceMapper(
         return filterIsInstance<ContentHolderTrait>().firstOrNull() ?: DefaultContentHolderTrait(null)
     }
 
-    private fun List<ExperienceTrait>.getExperiencePresentingTraitOrDefault(): PresentingTrait {
-        return filterIsInstance<PresentingTrait>().firstOrNull() ?: DefaultPresentingTrait(null, scope, context)
+    private fun List<ExperienceTrait>.getExperiencePresentingTraitOrThrow(): PresentingTrait {
+        return filterIsInstance<PresentingTrait>().firstOrNull() ?: throw AppcuesPresentingTraitNotFound()
     }
 
+    private class AppcuesPresentingTraitNotFound : Exception("Presenting capability trait required")
```

Comment:
I dont think so because this is an Exception during mapper, doesnt actually need to be a traitException, we got some others similar to this one like in [ActionRegestry](https://github.com/appcues/appcues-android-sdk/blob/ca958b7a5379c4b67777b65cd5877392151d9d73/appcues/src/main/java/com/appcues/action/ActionRegistry.kt#L49)

Reply:
ok, makes sense.